tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java

```diff
@@ -156,13 +155,13 @@ public void testXMLPILiteral() {
         Assert.assertEquals(returns[1].stringValue(), "<?foo 11?>");
 
         Assert.assertTrue(returns[2] instanceof BXML);
-        Assert.assertEquals(returns[2].stringValue(), "<?foo  aaa11bbb22ccc?>");
```

Comment:
why are the test changed?

Reply:
This is a bug in the old xml-parser, where ws is captured as part of the xml-pi `char*` section. This is fixed in the new parser, now any ws is capture as ws.

Now we are hitting a different bug which is there in BLangNodes, where ws is not propagated to runtime - BLandNodes ignore ws when creating XMLs. @KavinduZoysa shall we create and issue for this?

Reply:
This is a bug in the old xml-parser, where ws is captured as part of the xml-pi char* section. This is fixed in the new parser, now any ws is capture as ws.

Now we are hitting a different bug which is there in BLangNodes, where ws is not propagated to runtime - BLandNodes ignore ws when creating XMLs. @KavinduZoysa  shall we create and issue for this?

Reply:
Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/24065